mirrord/sip/src/lib.rs

```diff
@@ -480,7 +480,15 @@ mod main {
             return Ok(NoSip);
         }
 
-        if let Some(shebang) = read_shebang_from_file(&complete_path)? {
+        if let Ok(option) = read_shebang_from_file(&complete_path) {
+            let shebang = match option {
+                Some(shebang) => shebang,
+                None => ScriptShebang {
```

Comment:
i think that non-text files will return an `Err` instead of an `Ok(Option)` here from `read_shebang_from_file()`, but maybe i'm wrong 👀 I'll test with binaries

Reply:
you were correct, this breaks the expected distrinction between SipScript and SipBinary 🙃 needs an additional check